src/concurrency.md

```diff
@@ -70,7 +71,102 @@ fn main() {
 }
 ```
 
+[ex-rayon-thumbnails]: #ex-rayon-thumbnails
+<a name="ex-rayon-thumbnails"></a>
+## Generate thumbnails for all .jpg in current directory
 
+[![rayon-badge]][rayon] [![glob-badge]][glob] [![image-badge]][image] [![cat-concurrency-badge]][cat-concurrency] [![cat-filesystem-badge]][cat-filesystem]
+
+This example uses the `glob`, `image`, and `rayon` crates to generate thumbnails for all .jpg files in the current directory. 
+The thumbnails are stored in a new folder called `thumbnails`.
+
+Files are found with [`glob::glob_with`] so we can match case insensitively on both `.jpg` and `.JPG`.
+Images are then resized in parallel with [`par_iter`] using [`DynamicImage::resize`].
+
+
+```rust
+extern crate glob;
+extern crate image;
+extern crate rayon;
+
+use std::path::{Path, PathBuf};
+use std::fs;
+use std::fs::File;
+
+use glob::{glob_with, MatchOptions};
+use image::{FilterType, ImageResult};
+use rayon::prelude::*;
+
+fn main() {
```

Comment:
we will want to be able to use `?` in our main so 
this function should be renamed to `fn run() -> Result<()>` and it will be called from `quick_main!(run);` as described in [a note about error handling](https://rust-lang-nursery.github.io/rust-cookbook/about.html#a-note-about-error-handling)